### PR TITLE
Set bt-dev-prod team as CODEOWNERS for all build scripts

### PR DESCRIPTION
**Pull request by blindpirate**
Closes https://github.com/gradle/gradle-private/issues/3710

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -75,3 +75,10 @@ subprojects/docs/src/docs/userguide/dep-man/
 subprojects/docs/src/docs/userguide/jvm/                                          @gradle/bt-jvm
 subprojects/docs/src/docs/userguide/reference/directory_layout.adoc               @gradle/bt-jvm
 subprojects/docs/src/docs/userguide/troubleshooting/version_catalog_problems.adoc @gradle/bt-jvm
+
+# Dev-prod team
+build-logic/                                @gradle/bt-developer-productivity
```
Comment (wolfs) [inline .github/CODEOWNERS:80]: 🤔 There are some things under `gradle` that we might want to add. Especially `mirrors.settings.gradle.kts`.

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -75,3 +75,10 @@ subprojects/docs/src/docs/userguide/dep-man/
...
+build-logic-commons/                        @gradle/bt-developer-productivity
+build-logic-settings/                       @gradle/bt-developer-productivity
+subprojects/*/build.gradle*                 @gradle/bt-developer-productivity
+*.gradle.kts                                @gradle/bt-developer-productivity
```
Comment (wolfs) [inline .github/CODEOWNERS:84]: 🤔 The `.teamcity` project should be added here. I wonder about `.github/workflows` as well.

Comment (blindpirate) [reply .github/CODEOWNERS:87]: Added:

```
.teamcity/                                  @gradle/bt-developer-productivity
.github/                                    @gradle/bt-developer-productivity
gradle/shared-with-buildSrc                 @gradle/bt-developer-productivity
```

File: .github/CODEOWNERS [hunk 3]
```diff
@@ -75,3 +75,13 @@ subprojects/docs/src/docs/userguide/dep-man/
 subprojects/docs/src/docs/userguide/jvm/                                          @gradle/bt-jvm
 subprojects/docs/src/docs/userguide/reference/directory_layout.adoc               @gradle/bt-jvm
 subprojects/docs/src/docs/userguide/troubleshooting/version_catalog_problems.adoc @gradle/bt-jvm
+
+# Dev-prod team
+.teamcity/                                  @gradle/bt-developer-productivity
+.github/                                    @gradle/bt-developer-productivity
+gradle/shared-with-buildSrc                 @gradle/bt-developer-productivity
```
Comment (wolfs) [inline .github/CODEOWNERS:82]:
```suggestion
gradle/shared-with-buildSrc/                @gradle/bt-developer-productivity
```

Is there a `/` missing at the end here? What is the consequence of leaving it off?

**Review by wolfs — APPROVED** [review]

File: .github/CODEOWNERS (hunk 3 above)
Comment (blindpirate) [reply .github/CODEOWNERS]: No idea, but let's add it.

**Comment by blindpirate** [comment]
@bot-gradle test and merge

**Comment by blindpirate** [comment]
@bot-gradle test and merge

**Pull request merged**